File: 2.两数相加.java
/*
 * @lc app=leetcode.cn id=2 lang=java
 *
 * [2] 两数相加
 */

// @lc code=start
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode
 * next; ListNode(int x) { val = x; } }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        // return root.next
        ListNode root = new ListNode(0);
        ListNode cur = root;
        int nex = 0;
        while (!(l1 == null && l2 == null && nex == 0)) {
            int l1_val = l1 != null ? l1.val : 0;
            int l2_val = l2 != null ? l2.val : 0;
            int sum = l1_val + l2_val + nex;
            cur.next = new ListNode(sum % 10);
            cur = cur.next;
            nex = sum / 10;
            l1 = l1 != null ? l1.next : null;
            l2 = l2 != null ? l2.next : null;
        }
        return root.next;

    }
}
// @lc code=end

File: 2.两数相加.py
#
# @lc app=leetcode.cn id=2 lang=python3
#
# [2] 两数相加
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        root = ListNode(0)
        cur = root
        nex = 0
        while not (l1 == None and l2 == None and nex == 0):
            l1_val = l1.val if l1 != None else 0
            l2_val = l2.val if l2 != None else 0
            sum_ = l1_val + l2_val + nex
            cur.next = ListNode(sum_ % 10)
            cur = cur.next
            nex = sum_ // 10
            l1 = l1.next if l1 != None else None
            l2 = l2.next if l2 != None else None

        return root.next
# @lc code=end
